workflow/controller/operator.go

```diff
@@ -913,7 +913,7 @@ func (woc *wfOperationCtx) processNodeRetries(node *wfv1.NodeStatus, retryStrate
 		}
 
 		// See if we have waited past the deadline
-		if time.Now().Before(waitingDeadline) {
+		if time.Now().Before(waitingDeadline) && int32(len(node.Children)) <= retryStrategy.Limit.IntVal {
```

Comment:
Could you elaborate a bit on the logic here? I can't see how this would fix the issue. Also, this needs tests

Reply:
Can you please add a test?

Reply:
@alexec I'm not quiet sure what I'm supposed to change. Do I need to change something in this function?

https://github.com/argoproj/argo-workflows/blob/6855f4c51b5bd667599f072ae5ddde48967006f1/workflow/controller/operator_test.go#L723-L774

Reply:
@rohankmr414, were you able to write a test to reproduce the issue?

Reply:
@blkperl I'm running this workflow and it fails when retryStrategy.limit is reached without any unnecessary backoff after the final retry. Is there a way we can get the backoff count for tests?

```yaml
metadata:
  generateName: test-backoff-
spec:
  entrypoint: main
  templates:
    - name: main
      retryStrategy:
        limit: 2
        backoff:
          duration: 2s
          factor: 2
          maxDuration: 5m
      container:
        name: main
        image: 'argoproj/argosay:v2'
        args: [ exit, "1" ]
```

Reply:
` retryStrategy.Limit.IntVal` can be zero, I think you need to use a func.

This PR needs to have a test added to it please.

Reply:
done